### PR TITLE
Implement leaderboad overlay at round and game end.

### PR DESCRIPTION
**Pull request by alecpm**
Implementation of ScrumDo story 230. Provides new config parameters `leaderboard_group`, `leaderboard_individual`, and `leaderboard_time`. At the end of each round (and at game end) a leaderboard overlay will be displayed for `leaderboard_time` seconds. The next round will not start until the interval has ended at which point the leaderboard is hidden.

### CODE REVIEW

**Comment by jessesnyder** [comment]
Currently, the leaderboard only shows the player's names when `leaderboard_individual` is set. Since I may not know my player's name, could we show something like: "You (Susie Smith): 8" or "Susie Smith (you): 8"?

File: dlgr/griduniverse/static/scripts/demo.js [hunk 1]
```diff
@@ -744,6 +776,67 @@ function onGameStateChange(msg) {
...
+    return;
+  }
+  var $score_list;
+  var $board_overlay = $('#leaderboard');
+  var group_scores = {};
+  var group_order = [];
+  $board_overlay.empty();
+  if (settings.leaderboard_group) {
+    players.each(function (i, player) {
+      if (settings.leaderboard_group) {
```
Comment (jessesnyder) [inline dlgr/griduniverse/static/scripts/demo.js]: Same condition already exists 2 lines above

File: dlgr/griduniverse/static/scripts/demo.js [hunk 2]
```diff
@@ -744,6 +776,67 @@ function onGameStateChange(msg) {
...
+        var color_name = color2name(player.color);
+        var cur_score = group_scores[color_name] || 0;
+        group_scores[color_name] = cur_score + Math.round(player.score);
+      }
+    });
+    group_order = Object.keys(group_scores).sort(function (a, b) {
+      return group_scores[a] > group_scores[b] ? -1 : (group_scores[a] < group_scores[b] ? 1 : 0);
+    });
+  }
+  if (settings.leaderboard_group) {
```
Comment (jessesnyder) [inline dlgr/griduniverse/static/scripts/demo.js]: Should we have just 1 occurrence of this condition? (Currently 2)

Comment (alecpm) [inline dlgr/griduniverse/static/scripts/demo.js]: Yeah, silly refactoring mistake. Fixed now.

**Comment by alecpm** [comment]
I've re-implemented this to use the chat window to render the leaderboards. Additionally, I've made the refactoring that @jessesnyder suggested, moving the score calculation and sorting into the `PlayerSet`.

**Comment by alecpm** [comment]
Rebased to current master.

**Comment by suchow** [comment]
screenshot 2017-08-07 00 30 13

**Pull request merged**